### PR TITLE
Feature/worn tyres

### PR DESCRIPTION
**Pull request by alexsorindrago**
added count for how many worn tyres are in a storage point

### CODE REVIEW

File: src/main/java/com/bucur/repository/HotelRepositoryInMemory.java [hunk 1]
```diff
@@ -9,17 +10,20 @@
 
     private List<StoragePoint> storagePoints = new ArrayList<>();
 
-    public HotelRepositoryInMemory() {
-//        setup();
-    }
+    Tyre tyre = new Tyre();
```
Comment (cosminbucur) [inline src/main/java/com/bucur/repository/HotelRepositoryInMemory.java]: what's the point of this tyre? and why is it instantiated?

File: src/main/java/com/bucur/repository/HotelRepositoryInMemory.java [hunk 2]
```diff
@@ -29,7 +33,26 @@ public StoragePoint getStoragePoint(String licensePlate) {
...
+
     public void setStoragePoints(List<StoragePoint> storagePoints) {
         this.storagePoints = storagePoints;
     }
+
+    public void setTyres(List<Tyre> tyres) {
+        this.tyres = tyres;
+    }
+
+    public long countWornTyres(HotelRepositoryInMemory storagePoint) {
```
Comment (cosminbucur) [inline src/main/java/com/bucur/repository/HotelRepositoryInMemory.java]: 1. the number of tyres is an integer number. no point in using long.
2. the parameter storagePoint has a misleading name. it is actually a repository not a storage point.
3. the method countWornTyres is on the HotelRepositoryInMemory object. why would you want to use an instance of the same object as parameter?

File: src/test/java/com/bucur/service/HotelServiceTest.java [hunk 3]
```diff
@@ -15,42 +12,39 @@
...
+        wornTyre2.wear = 1;
+        newTyre1.wear = 6;
+        wornTyre3.wear = 1;
+
+        HotelRepositoryInMemory tyresInStoragePoint = new HotelRepositoryInMemory();
+        tyresInStoragePoint.setTyres(Arrays.asList(wornTyre1, wornTyre2, newTyre1, wornTyre3));
         // when
         // on when, we call the method under test from the responsible object
-        int actualResult = hotelService.countWornTyres();
+        long actualResult = tyresInStoragePoint.countWornTyres(tyresInStoragePoint);
```
Comment (cosminbucur) [inline src/test/java/com/bucur/service/HotelServiceTest.java]: it is not clear that tyresInStoragePoint is actually a repository

File: src/main/java/com/bucur/repository/HotelRepositoryInMemory.java [hunk 4]
```diff
@@ -7,13 +7,29 @@
...
+        storagePoint2.licensePlate = "B";
+    }
 
     @Override
     public StoragePoint getStoragePoint(String licensePlate) {
         return storagePoints.stream()
-            .filter(storagePoint -> storagePoint.licensePlate.equals(licensePlate))
-            .findFirst()
-            .orElseThrow(() -> new RuntimeException("no tyres found for license plate " + licensePlate));
+                .filter(storagePoint -> storagePoint.licensePlate.equals(licensePlate))
```
Comment (cosminbucur) [inline src/main/java/com/bucur/repository/HotelRepositoryInMemory.java]: do not commit wrong formatting and code alignment

File: src/main/java/com/bucur/repository/HotelRepositoryInMemory.java [hunk 5]
```diff
@@ -33,26 +42,10 @@ public StoragePoint getStoragePoint(String licensePlate) {
...
-        return count;
-
-//        int minimumWearIndicator=3;
-//        return tyres.stream()
-//                .filter(tyre -> tyre.wear < minimumWearIndicator)
-//                .findFirst()
-//                .orElseThrow(() -> new RuntimeException("no worn tyres found"));
+        if (count > 0) {
+            return count;
+        } else throw new RuntimeException("no used tyre for storage point " + storagePoint);
```
Comment (cosminbucur) [inline src/main/java/com/bucur/repository/HotelRepositoryInMemory.java]: the repository is not a storage point

File: src/main/java/com/bucur/repository/HotelRepositoryInMemory.java (hunk 1 above)
Comment (alexsorindrago) [reply src/main/java/com/bucur/repository/HotelRepositoryInMemory.java]: no point. deleted.

File: src/main/java/com/bucur/repository/HotelRepositoryInMemory.java (hunk 2 above)
Comment (alexsorindrago) [reply src/main/java/com/bucur/repository/HotelRepositoryInMemory.java]: changed long to int, type and name of parameter

File: src/main/java/com/bucur/repository/HotelRepositoryInMemory.java (hunk 4 above)
Comment (alexsorindrago) [reply src/main/java/com/bucur/repository/HotelRepositoryInMemory.java]: dude you wrote that part :))

File: src/main/java/com/bucur/repository/HotelRepositoryInMemory.java (hunk 5 above)
Comment (alexsorindrago) [reply src/main/java/com/bucur/repository/HotelRepositoryInMemory.java]: fixed

**Pull request closed without merging**